pkg/machine/provider/platform_windows.go

```diff
@@ -39,17 +45,26 @@ func GetByVMType(resolvedVMType define.VMType) (vmconfigs.VMProvider, error) {
 	case define.WSLVirt:
 		return new(wsl.WSLStubber), nil
 	case define.HyperVVirt:
+		// Check permissions before returning the Hyper-V provider.
+		// Working with Hyper-V requires users to be at least members of the Hyper-V admin group.
+		// Init and remove actions have custom use cases and they are checked on the stubber.
+		if !hasAdminRightsFunc() && !hasHyperVAdminRightsFunc() {
+			return nil, hyperv.ErrHypervUserNotInAdminGroup
+		}
 		return new(hyperv.HyperVStubber), nil
 	default:
 	}
 	return nil, fmt.Errorf("unsupported virtualization provider: `%s`", resolvedVMType.String())
 }
 
 func GetAll() []vmconfigs.VMProvider {
-	return []vmconfigs.VMProvider{
+	providers := []vmconfigs.VMProvider{
 		new(wsl.WSLStubber),
-		new(hyperv.HyperVStubber),
 	}
+	if hasAdminRightsFunc() || hasHyperVAdminRightsFunc() {
```

Comment:
I am struggling with the logic here if I am a user.  If I don't have permissions, should it not exist?  Earlier, when we returned an error, that certainly made sense to me.

Reply:
This is about `podman machine ls`, and the problem has changed in 6.0. If the user doesn't have privileges we still want to list WSL machines.

But we should indeed fail when the user specifies `--provider hyperv`. And if the default provider is `hyperv` too.